testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/policies/ClientPoliciesTest.java

```diff
@@ -1223,4 +1225,54 @@ private void testProhibitedImplicitOrHybridFlow(boolean isOpenid, String respons
         assertEquals(expectedError, oauth.getCurrentFragment().get(OAuth2Constants.ERROR));
         assertEquals(expectedErrorDescription, oauth.getCurrentFragment().get(OAuth2Constants.ERROR_DESCRIPTION));
     }
+
+    @Test
+    public void testSecureParContentsExecutor() throws Exception {
+        // register profiles
+        String json = (new ClientProfilesBuilder()).addProfile(
+                (new ClientProfileBuilder()).createProfile(PROFILE_NAME, "Le Premier Profil")
+                        .addExecutor(SecureParContentsExecutorFactory.PROVIDER_ID, null)
+                        .toRepresentation()
+        ).toString();
+        updateProfiles(json);
+
+        String clientBetaId = generateSuffixedName("Beta-App");
+        createClientByAdmin(clientBetaId, (ClientRepresentation clientRep) -> {
+            clientRep.setSecret("secretBeta");
+        });
+
+        // register policies
+        json = (new ClientPoliciesBuilder()).addPolicy(
+                (new ClientPolicyBuilder()).createPolicy(POLICY_NAME, "La Premiere Politique", Boolean.TRUE)
+                        .addCondition(AnyClientConditionFactory.PROVIDER_ID,
+                                createAnyClientConditionConfig())
+                        .addProfile(PROFILE_NAME)
+                        .toRepresentation()
+        ).toString();
+        updatePolicies(json);
+
+        // Pushed Authorization Request
+        ParResponse pResp = oauth.doPushedAuthorizationRequest(clientBetaId, "secretBeta");
+        assertEquals(201, pResp.getStatusCode());
+        String requestUri = pResp.getRequestUri();
+
+        try {
+            oauth.requestUri(requestUri);
+            oauth.clientId(clientBetaId);
+            oauth.openLoginForm();
+            assertTrue(errorPage.isCurrent());
+            assertEquals("PAR request did not include necessary parameters", errorPage.getError());
+
+            oauth.requestUri(null);
+            pResp = oauth.doPushedAuthorizationRequest(clientBetaId, "secretBeta");
+            assertEquals(201, pResp.getStatusCode());
+            requestUri = pResp.getRequestUri();
+            oauth.requestUri(requestUri);
+
+            oauth.stateParamHardcoded(null);
+            successfulLoginAndLogout(clientBetaId, "secretBeta");
+        } catch (Exception e) {
+            fail();
```

Comment:
Minor: Is it any point in this try/catch block? IMO it can be better to remove it (and also the call to `fail()`).

I see that `fail()` will just throw AssertionError without telling any details what's wrong and why the test fail, so doesn't look like very good thing to use IMO...

Reply:
Thank you. Yes, I agree with you. I will remove it.

Reply:
I have removed it.